specs/TrustedOriginSetting.md

```diff
@@ -0,0 +1,205 @@
+  /// SetTrustedOriginFeatures to configure features for trusted origins.
+  HRESULT CreateTrustedOriginFeatureSettings(
+      [in] UINT32 featuresCount,
+      [in] COREWEBVIEW2_TRUSTED_ORIGIN_FEATURE* features,
+      [in] BOOL* isEnabled
+      , [out, retval] ICoreWebView2StagingTrustedOriginFeatureSettingCollectionView** value);
+
+  /// Sets the feature configurations for specified origins.
+  /// This method allows configuring multiple features for trusted origins,
+  /// such as accent color, persistent storage, and enhanced security mode.
+  /// The origins can be both exact origin strings and wildcard patterns.
+  /// For detailed examples, refer to the table at: https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter.
+  HRESULT SetTrustedOriginFeatures(
+      [in] UINT32 originsCount,
+      [in] LPCWSTR* origins,
+      [in] ICoreWebView2StagingTrustedOriginFeatureSettingCollectionView* features
+  );
+
+  /// Gets the feature configurations for a specified origin.
+  /// Returns a collection of feature settings that have been configured for the origin.
+  /// If no features have been configured for the origin, an empty collection is returned.
+  /// The origin can be both exact origin strings and wildcard patterns.
+  /// For detailed examples, refer to the table at: https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.addwebresourcerequestedfilter.
+  HRESULT GetTrustedOriginFeatures(
+      [in] LPCWSTR origin
+      , [in] ICoreWebView2StagingGetTrustedOriginFeaturesCompletedHandler* handler);
+}
+
+/// Represents a feature setting configuration for a trusted origin.
+[uuid(edf2c30e-daab-572c-887b-61e5acb8c305), object, pointer_default(unique)]
+interface ICoreWebView2StagingTrustedOriginFeatureSetting : IUnknown {
+  /// The feature type for this setting.
+  [propget] HRESULT Feature([out, retval] COREWEBVIEW2_TRUSTED_ORIGIN_FEATURE* value);
+
+  /// Indicates whether the feature is enabled for the origin.
+  [propget] HRESULT IsEnabled([out, retval] BOOL* value);
+}
+
+
+/// A collection of trusted origin settings.
+interface ICoreWebView2StagingTrustedOriginFeatureSettingCollectionView : IUnknown {
+  /// Gets the number of objects contained in the `TrustedOriginFeatureSettingCollection`.
+  [propget] HRESULT Count([out, retval] UINT32* value);
+
+  /// Gets the object at the specified index.
+  HRESULT GetValueAtIndex(
+      [in] UINT32 index
+      , [out, retval] ICoreWebView2StagingTrustedOriginFeatureSetting** value);
+}
+```
+
+## .Net/WinRT
+
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    public enum CoreWebView2TrustedOriginFeature
+    {
+        AccentColor = 0,
+        PersistentStorage = 1,
+        EnhancedSecurityMode = 2,
+    }
+
+    public partial class CoreWebView2Profile
+    {
+        public async Task GetTrustedOriginFeaturesAsync(string origins);

```

Comment:
The return value of this should be like IEnumerable<KeyValuePair<... or something right? Not just Task?